koku/api/report/serializers.py

```diff
@@ -324,11 +324,13 @@ def validate(self, data):
         """
         super().validate(data)
 
-        if data.get("category") and ("key_only" not in data.keys()):
-            # allow category to be passed in without group by for the tags endpoint
-            if data.get("category") and (not data.get("group_by") or not data.get("group_by").get("project")):
-                error = {"error": ("Category may not be used without a group_by project parameter")}
-                raise serializers.ValidationError(error)
+        if data.get("category") and ("key_only" in data):
+            error = {"error": "Category may only be used as a filter for the tag endpoints."}
+            raise serializers.ValidationError(error)
+
+        if data.get("category") and (not data.get("group_by") or not data.get("group_by").get("project")):
+            error = {"error": "Category may not be used without a group_by project parameter"}
+            raise serializers.ValidationError(error)
```

Comment:
Moving the overlapping condition to the top makes this a bit easier to read.
```suggestion
        if data.get("category")
            if "key_only" in data:
                error = {"error": "Category may only be used as a filter for the tag endpoints."}
                raise serializers.ValidationError(error)

             if not data.get("group_by") or not data.get("group_by").get("project"):
                error = {"error": "Category may not be used without a group_by project parameter"}
                raise serializers.ValidationError(error)
```

I was also thinking the second `data.get("group_by")` should return `{}` rather than `None` to avoid a potential `AttributeError`, but an "or" comparison will short circuit on the first false condition, so I think it should be ok to leave that as is.